src/Compilers/CSharp/Portable/Syntax/SyntaxKindFacts.cs

```diff
@@ -352,6 +336,22 @@ public static bool IsNamespaceMemberDeclaration(SyntaxKind kind)
                 case SyntaxKind.InterfaceDeclaration:
                 case SyntaxKind.DelegateDeclaration:
                 case SyntaxKind.EnumDeclaration:
+                    return true;
+
+                default:
+                    return false;
+            }
+        }
+
+        public static bool IsNamespaceMemberDeclaration(SyntaxKind kind)
+        {
+            if (IsTypeDeclaration(kind))
+            {
+                return true;
+            }
+
+            switch (kind)
+            {
                 case SyntaxKind.NamespaceDeclaration:
                     return true;
                 default:
```

Comment:
would love to just make this `=> IsTypeDeclaration(kind) || kind == Namespace`

Reply:
Beautiful. I was looking at that before I pushed wondering how to make it better...